src/components/select-user.tsx

```diff
@@ -0,0 +1,134 @@
+import { Fragment, useState } from "react";
+import Image from "next/image";
+import { faXmark } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { Listbox } from "@headlessui/react";
+import { useQuery } from "@tanstack/react-query";
+
+import HoverOpacityButton from "@/components/Button/hover-opacity-button";
+import Select from "@/components/select";
+import LoadingSpinner from "@/components/UI/loading-spinner";
+import { useUsers } from "@/hooks/users";
+import { httpClient } from "@/lib/base-http-client";
+import cn from "@/lib/classnames";
+import isBlank from "@/lib/is-blank";
+import { User } from "@/types";
+const NAME_KEY = "emailAddress";
+const VALUE_KEY = "id";
+interface SelectUserProps {
+  value?: User[];
+  onChange?: (value: User[]) => void;
+  multiple?: boolean;
+  label?: string;
+  initialUserIds?: string[];
+}
+
+// todo: improve UX
+// todo: add dynamic loading when scroll to bottom
```

Comment:
For next time, take a look at https://tanstack.com/query/latest/docs/react/reference/useInfiniteQuery